mitmproxy/protocol/http.py

```diff
@@ -359,7 +364,10 @@ def establish_server_connection(self, flow):
                 self.set_server(address, tls, address.host)
             # Establish connection is neccessary.
             if not self.server_conn:
-                self.connect()
```

Comment:
Yes. We should talk about this later or tomorrow, I actually disagree with all changes to `http.py`. :innocent: